1.1.3 analysers

van800 opened the pull request.

miniwolf committed:
Rewrite tests.

miniwolf committed:
Move main package to main package namespace.

miniwolf committed:
Rewrite tests using ProjectGenerationTestBase

van800 committed:
Improve projectGeneration to support analyzer, additionalfile and ruleset

van800 committed:
avoid empty lines in generated csproj

van800 committed:
fix csproj generation for multiple nowarn

van800 committed:
Merge remote-tracking branch 'origin/rewrite-tests' into 1.1.3-analysers

# Conflicts:
#	Packages/com.unity.ide.rider/Rider/Editor/ProjectGeneration/ProjectGeneration.cs

van800 committed:
fix merge

van800 committed:
cleanup using

JurjenBiewenga committed:
Warning levels defined in the csc file should now propagate to the csproject files

JurjenBiewenga commented on Packages/com.unity.ide.rider/Rider/Editor/ProjectGeneration.cs:
```diff
@@ -820,6 +826,58 @@ string SolutionText(IEnumerable<Assembly> islands)
+    {
+      var paths = responseFilesData.SelectMany(x =>
+      {
+        string start = $"/{key}:";
```
Keep in mind that the arguments do not have to start with a `/`; `-` is valid as well

van800 replied:
Nice catch, thanks

van800 committed:
refactoring getting additional csc.rsp data

van800 committed:
revert

JurjenBiewenga committed:
Merge 1.1.3-analysers

JurjenBiewenga committed:
Warn level now uses the new format

JurjenBiewenga committed:
Added test for warning level

JurjenBiewenga committed:
Added test checking whether the default warning level is correctly set

van800 committed:
Merge pull request #1 from JurjenBiewenga/csc-warn-level

Warning levels defined in the csc file should now propagate to the csproject files

JurjenBiewenga committed:
Added tests for analyzers/additional files/rulesets

Fixed analyzers and additional files delimited by `;`

van800 committed:
Merge pull request #54 from unity/1.1.3-analysers-van800

 Warning levels defined in the csc file should now propagate to the csproj

JurjenBiewenga committed:
Fixed empty lines in csprojectgeneration

Replace Object with the type keyword
Added testcase for multiple ruleset definitions

JurjenBiewenga committed:
Removed unused usings

JurjenBiewenga committed:
Replaced Environment.NewLine with "\r\n"

van800 committed:
Merge pull request #3 from JurjenBiewenga/AnalyzerTests

Analyzer tests and fixes for analyzer/additionalfiles/rulesets

JurjenBiewenga committed:
Added warnaserror support

van800 committed:
Merge remote-tracking branch 'origin/1.1.3' into 1.1.3-analysers

van800 committed:
Merge remote-tracking branch 'com.unity.ide.rider.van800/1.1.3-analysers' into 1.1.3-analysers

van800 committed:
add test for nowarn in csc.rsp

van800 committed:
remove setting, which allowed overriding LangVersion. Add ability to override langversion via csc.rsp

JurjenBiewenga committed:
Merge branch '1.1.3-analysers' of github.com:van800/com.unity.ide.rider into WarnAsError

# Conflicts:
#	Packages/com.unity.ide.rider/Rider/Editor/ProjectGeneration/ProjectGeneration.cs

JurjenBiewenga committed:
Added -doc support

van800 committed:
Merge pull request #4 from JurjenBiewenga/WarnAsError

Added warnaserror support
